### PR TITLE
Feature: Linter setup for Typescript

### PR DESCRIPTION
**Pull request by 0xOptimism**

### CODE REVIEW

**Comment by ken333135** [comment]
does this linter work as expected? i havent tried it out

File: .husky/pre-commit [hunk 1]
```diff
@@ -2,4 +2,5 @@
 echo "pre-commit"
 . "$(dirname "$0")/_/husky.sh"
 
-npx lint-staged --allow-empty
\ No newline at end of file
+npx lint-staged --allow-emptyyarn tsc --noEmit && yarn eslint . && yarn prettier --write .
+yarn lint-staged .
```
Comment (sunshine1004-dev) [inline .husky/pre-commit:6]: Maybe we can add `yarn build` here to check code styles for production mode.

Comment (0xOptimism) [reply .husky/pre-commit:6]: Sound good @sunshine1004-dev

**Comment by 0xOptimism** [comment]
> does this linter work as expected? i havent tried it out

@ken333135 I just add to add a few thing in here, i will ping you when i'm done, would be nice if you could test it out to make sure it's not only working on my local

**Pull request merged**